gprotorch/kernels/bit_vector_kernel.py

```diff
@@ -0,0 +1,181 @@
+"""
+Module for kernels that operate on bit vectors.
+Author: Ryan-Rhys Griffiths 2021
+"""
+
+import gpytorch
+from gpytorch.kernels import Kernel
+from gpytorch.kernels.kernel import default_postprocess_script
+import torch
+
+
+class BitDistance(torch.nn.Module):
+    r"""
+    Distance module for bit vector kernels.
+    """
+    def __init__(self, postprocess_script=default_postprocess_script):
+        super().__init__()
+        self._postprocess = postprocess_script
+
+    def _sim(self, x1, x2, postprocess, x1_eq_x2=False, metric='tanimoto'):
+        r"""
+        Computes the similarity between x1 and x2
+
+        Args:
+            :attr: `x1`: (Tensor `n x d` or `b x n x d`):
+                First set of data where b is a batch dimension
+            :attr: `x2`: (Tensor `m x d` or `b x m x d`):
+                Second set of data where b is a batch dimension
+            :attr: `postprocess` (bool):
+                Whether to apply a postprocess script (default is none)
+            :attr: `x1_eq_x2` (bool):
+                Is x1 equal to x2
+            :attr: `metric` (str):
+                String specifying the similarity metric. One of ['tanimoto']
+
+        Returns:
+            (:class:`Tensor`, :class:`Tensor) corresponding to the similarity matrix between `x1` and `x2`
+        """
+
+        if metric not in ['tanimoto']:
+            raise RuntimeError('Similarity metric not supported. Available options are \'tanimoto\'')
+
+        if metric == 'tanimoto':
```

Comment:
Do we need this check? Looks like we're already raising an error if this isn't the case

Reply:
So this branch will take us into the Tanimoto metric, I'll add further branches for the other metrics we introduce